lib/execjs/graaljs_runtime.rb

```diff
@@ -0,0 +1,144 @@
+require "execjs/runtime"
+
+module ExecJS
+  class GraalJSRuntime < Runtime
+    class Context < Runtime::Context
+      def initialize(runtime, source = "", options = {})
+        @context = Polyglot::InnerContext.new
+        @context.eval('js', 'delete this.console')
+        @js_object = @context.eval('js', 'Object')
+
+        source = encode(source)
+        unless source.empty?
+          translate do
+            eval_in_context(source)
+          end
+        end
+      end
+
+      def exec(source, options = {})
+        source = encode(source)
+        source = "(function(){#{source}})()" if /\S/.match?(source)
+
+        translate do
+          eval_in_context(source)
+        end
+      end
+
+      def eval(source, options = {})
+        source = encode(source)
+        source = "(#{source})" if /\S/.match?(source)
+
+        translate do
+          eval_in_context(source)
+        end
+      end
+
+      def call(source, *args)
+        source = encode(source)
+        source = "(#{source})" if /\S/.match?(source)
+
+        translate do
+          function = eval_in_context(source)
+          function.call(*convert_ruby_to_js(args))
+        end
+      end
+
+      private
+
+      def translate
+        begin
```

Comment:
nitpick: the `begin/end` could be implicit

Reply:
Indeed, I thought that was somewhat recently introduced but it seems it's been there for methods forever, only for `do/end` it was introduced in 2.5.

Reply:
done